File: app/src/main/java/org/cuieney/videolife/di/module/ActivityModule.java
package org.cuieney.videolife.di.module;

import android.app.Activity;


import org.cuieney.videolife.di.PerActivity;

import dagger.Module;
import dagger.Provides;

/**
 * 这个module把activity暴露给相关联的类。
 * 比如在fragment中使用activity的context。
 */
@Module
public class ActivityModule {
    private Activity mActivity;

    public ActivityModule(Activity activity) {
        this.mActivity = activity;
    }

    @Provides
    @PerActivity
    public Activity provideActivity() {
        return mActivity;
    }

}

File: app/src/main/java/org/cuieney/videolife/kotlin/di/module/ActivityModule.kt
package org.cuieney.videolife.kotlin.di.module

import android.app.Activity
import dagger.Module
import dagger.Provides
import org.cuieney.videolife.di.PerActivity

/**
 * Created by cuieney on 2017/5/19.
 */
@Module class ActivityModule constructor(activity: Activity){
    var act : Activity = activity

    @Provides
    @PerActivity
    fun provideActivity(): android.app.Activity = act

}